packages/api/tests/lib/rules/test-updateRuleResources.js

```diff
@@ -0,0 +1,475 @@
+    await deleteRuleResources(testKnex, updatedRule);
+  });
+});
+
+test.serial('Updating a kinesis type rule workflow does not affect value or event source mappings', async (t) => {
+  const { testKnex } = t.context;
+  const kinesisRule = fakeRuleRecordFactory({
+    type: 'kinesis',
+    workflow,
+    enabled: true,
+    value: randomString(),
+  });
+  // create rule trigger
+  const createdRule = await createRuleTrigger(kinesisRule);
+  const kinesisEventMappings = await getKinesisEventMappings();
+  const consumerEventMappings = kinesisEventMappings[0].EventSourceMappings;
+  const logEventMappings = kinesisEventMappings[1].EventSourceMappings;
+
+  t.is(consumerEventMappings.length, 1);
+  t.is(logEventMappings.length, 1);
+  t.is(consumerEventMappings[0].UUID, createdRule.arn);
+  t.is(logEventMappings[0].UUID, createdRule.log_event_arn);
+
+  // update rule workflow
+  const updatedKinesisRule = {
+    ...createdRule,
+    workflow: 'new-workflow',
+  };
+  const updatedRule = await updateRuleTrigger(kinesisRule, updatedKinesisRule, testKnex);
+  const updatedKinesisEventMappings = await getKinesisEventMappings();
+  const updatedconsumerEventMappings = updatedKinesisEventMappings[0].EventSourceMappings;
+  const updatedlogEventMappings = updatedKinesisEventMappings[1].EventSourceMappings;
+
+  t.is(updatedconsumerEventMappings.length, 1);
+  t.is(updatedlogEventMappings.length, 1);
+  t.is(updatedconsumerEventMappings[0].UUID, updatedRule.arn);
+  t.is(updatedlogEventMappings[0].UUID, updatedRule.log_event_arn);
+
+  t.is(createdRule.arn, updatedRule.arn);
+  t.is(createdRule.log_event_arn, updatedRule.log_event_arn);
+
+  // Clean Up
+  t.teardown(async () => {
+    await deleteRuleResources(testKnex, createdRule);
+    await deleteRuleResources(testKnex, updatedRule);
+  });
+});
+
+test.serial('Updating a valid SQS rule to have an invalid schema throws an error and does not update triggers', async (t) => {
+  const queues = await createSqsQueues(randomString());
+  const sqsRule = fakeRuleRecordFactory({
+    workflow,
+    type: 'sqs',
+    value: queues.queueUrl,
+    enabled: true,
+    meta: {
+      visibilityTimeout: 100,
+      retries: 4,
+    },
+  });
+  const sqsRuleWithTrigger = await createRuleTrigger(sqsRule);
+  t.is(sqsRuleWithTrigger.value, sqsRule.value);
+
+  // update rule to be invalid by setting type to null
+  const updatedKinesisRule = {
+    ...sqsRuleWithTrigger,
+    type: null,
+  };
+  await t.throwsAsync(
+    updateRuleTrigger(sqsRuleWithTrigger, updatedKinesisRule, t.context.testKnex),
+    { name: 'SchemaValidationError' }
+  );
+  const updatedKinesisEventMappings = await getKinesisEventMappings();
+  const updatedConsumerEventMappings = updatedKinesisEventMappings[0].EventSourceMappings;
+  const updatedLogEventMappings = updatedKinesisEventMappings[1].EventSourceMappings;
+
+  t.is(updatedConsumerEventMappings.length, 0);
+  t.is(updatedLogEventMappings.length, 0);
```

Comment:
None of this is relevant for SQS rules

Reply:
[`aa59c90` (#2871)](https://github.com/nasa/cumulus/pull/2871/commits/aa59c90a42ee176d4075b4de2c1bf78678177b09#diff-057c67806291c57c6410d819b57bda8d87b5370152a299e6baca44b5a44f0e40L299) 🤦